src/utils/taskActions.js

```diff
@@ -21,13 +21,16 @@ export const addTask = (newTask, setTasks, tasks, setError, resetModal) => {
     return;
   }
 
-  setTasks([
+  const updatedTasks = [
     ...tasks,
     {
       ...newTask,
       taskId: uuid(),
     },
-  ]);
+  ];
+  setTasks(updatedTasks);
+
+  localStorage.setItem("userTasks", JSON.stringify(updatedTasks));
```

Comment:
Suggestion, you could also try implementing this within a useEffect inside your context, but you will have to pass the updated state within context